conda/cli/main.py

```diff
@@ -147,6 +147,58 @@ def main(*args, **kwargs):
             return ExceptionHandler().handle_exception(exc_val, exc_tb)
 
     from ..exceptions import conda_exception_handler
+
+    # Here we are detecting (badly) the case when conda.bat has been run but not
+    # to activate or deactivate. conda.bat will have added entries for sys.prefix
+    # to the front of PATH (so that this conda executable can find its DLLs), but
+    # that is problematic for conda run, if executed through conda.bat because
+    # we do not prefix PATH with the new prefixes entries; we swap them in-place
+    # (a-la reactivate) or may even do nothing at all when re-activating the
+    # currently active env.
+    #
+    # I do not want to be messing with PATH here at all though (at least not until
+    # conda has loaded every DLL it could possibly want). We need a new proxy for
+    # os.environ['PATH'] which gets used for any PATH qeuries made by conda and
+    # which strips sysp from the result (which would be emitted to all script files
+    # (that need to set the PATH env var) and all subprocess envs.
+    #
+
+    from os import environ, getpid, pathsep, sep
+    from psutil import Process
```

Comment:
This is a super heavy import, _right_ at the beginning of every single invocation of conda. We've got to figure out a way to be more frugal here.

Reply:
Super heavy and it doesn't even work! I am not sure I need it to work either, I think looking at `PATH` and maybe other env vars would be enough to determine whether conda has been run through `condabin\conda.bat` or not. Suggestions are welcome.

Note though that this is still WIP, but it's becoming pressing that I get this solved since I increasingly rely on conda run *during package builds* to test things (I can, to expedite things, comment out these tests though if you want this to remain in the oven for a while yet, it's certainly a bit half-baked at present; I am also interested in your thoughts on monkey-patching os.environ['PATH'] inside conda rather than me modifying the real thing directly as I am doing here).